blog/views.py

```diff
@@ -0,0 +1,15 @@
+from django.contrib.auth.mixins import LoginRequiredMixin
+from django.views.generic import ListView, DetailView
+from .models import Post
+
+
+class PostListView(ListView):
+    queryset = (Post.objects.all().order_by("-created_time")
+                .select_related("owner"))
+    paginate_by = 5
+    model = Post
+
+
+class PostDetailView(DetailView, LoginRequiredMixin):
```

Comment:
The LoginRequiredMixin should be placed before the DetailView in the PostDetailView class definition to ensure that the login requirement is enforced correctly.